polygeist/tools/cgeist/Lib/CGExpr.cc

```diff
@@ -1164,7 +1231,18 @@ MLIRScanner::emitSYCLOps(const clang::Expr *Expr,
       if (!Op)
         Op = Builder.create<mlir::sycl::SYCLCallOp>(
             Loc, OptRetType, OptFuncType, Func->getNameAsString(), Name, Args);
+    } else {
+      auto *ND = llvm::dyn_cast_or_null<clang::NamespaceDecl>(
+          Func->getEnclosingNamespaceContext());
+      const auto *II = ND ? ND->getIdentifier() : nullptr;
+      if (!ND || (II && II->isStr("__spirv"))) {
+        // SPIR-V builtin functions are declared either in the global namespace
+        // or in __spirv.
+        const auto Loc = getMLIRLocation(Expr->getExprLoc());
+        Op = createSYCLBuiltinOp(Func, Loc);
```

Comment:
assert Op or handle !Op.

Reply:
That is handled [here](https://github.com/intel/llvm/blob/ef3ada9c11e9aaa363186c15322dec8a19aaec17/polygeist/tools/cgeist/Lib/CGCall.cc#L404)